nixos/modules/services/monitoring/prometheus/exporters/collectd.nix

```
@@ -61,6 +61,7 @@ in
     collectSettingsArgs = if (cfg.collectdBinary.enable) then ''
       -collectd.listen-address ${cfg.collectdBinary.listenAddress}:${toString cfg.collectdBinary.port} \
       -collectd.security-level ${cfg.collectdBinary.securityLevel} \
+      ${optionalString (cfg.colledtdBinary.authFile != null) "-collextd.auth-file ${cfg.collectdBinary.authFile}"} \
```

Have you tested this?

```suggestion
      ${optionalString (cfg.colledtdBinary.authFile != null) "-collectd.auth-file ${cfg.collectdBinary.authFile}"} \
```

---

```suggestion
      ${optionalString (cfg.collectdBinary.authFile != null) "-collextd.auth-file ${cfg.collectdBinary.authFile}"} \
```

---

```suggestion
      ${optionalString (cfg.collecdtdBinary.authFile != null) "-collectd.auth-file ${cfg.collectdBinary.authFile}"} \
```